apps/gitness/src/framework/context/ThemeContext.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect } from 'react'
+
+import { create } from 'zustand'
+import { persist } from 'zustand/middleware'
+
+import { ColorType, ContrastType, FullTheme, ModeType, ThemeState } from '@harnessio/ui/components'
+
+const useThemeStore = create<ThemeState>()(
+  persist(
+    set => ({
+      theme: 'system-std-std' as FullTheme, // Default theme
+      setTheme: (newTheme: FullTheme) => set({ theme: newTheme })
+    }),
+    {
+      name: 'canary-ui-theme' // LocalStorage key
+    }
+  )
+)
+
+export const useTheme = () => {
```

Comment:
Can we add the return type to keep typescript happy and auto-completing?

Reply:
Done